### ISSUE TITLE
running Java in macOS Siera

### ISSUE THREAD

**Issue by SPineapple** [Emotiv/community-sdk#279]
Hello Duy Tan,

Upon your request, I am adding here the problem I'm facing, which is I am not receiving any signals or any data from Emotiv Insight except when I printed in the screen the number of _state_ I got **.1536** , I thought this was the only case, but when I turned off the headset, and I run again the program, I got the same number, also the UserID number is not printing out in screen. I can't figure out where I went wrong.
I am considering to work on C++, but I don't want my effort to go waste so far, but it would be my last option. 

I have attached a screen shot of the output I am having, this output is showing both when the headset is on or off. 

This code is from Java Example EmoStateLogger : 
![screen shot 2017-12-14 at 7 45 55 am](https://user-images.githubusercontent.com/18646285/33979633-50690f16-e0b6-11e7-9009-95579db315f6.png)

**Comment by duytan90** [comment]
Hi @SPineapple, both in your cases the headset wasn't connected to your Mac, therefor the [error 0x600](http://emotiv.github.io/community-sdk/_iedk_error_code_8h.html#a9ff6ae0daaa6987da68594d3a5f212dd) (1536 decimal) happened.
Connecting to Mac/PC is different to connecting to the running program, so make sure the headset is connected to your Mac first. Using [Emotiv Universal Dongle](https://www.emotiv.com/product/emotiv-usb-receiver-universal-model/) is highly recommended for fast and stable connection. I guess you were trying with Bluetooth, which is a bit harder to connect than the Universal Emotiv Dongle. Check with other Emotiv Applications or [this app](https://itunes.apple.com/us/app/lightblue/id639944780?mt=12) to make sure you've got the headset's connected to your Mac before trying the examples.

**Comment by SPineapple** [comment]
Hello @duytan90 ,
I have successfully run Emotiv using Emotiv Universal Dongle as you recommended, I often got the NEUTRAL result more than anything else, but I think this shouldn't be a major issue because I have now encountered a larger problem when I disconnect,
I used : `Edk.INSTANCE.IEE_EngineDisconnect();` before exiting my window, and after that I get an error about core dump?

I have used Terminal and run `ulimit -c unlimited ` as it was suggested, but it didn't solve it. 
I am working on a GUI, which only display the Electrode status, and when I press a Cancel button it should let me exit this window and get me back to Main Menu without crashing the whole project I built. But instead I am getting the following error : 
![screen shot 2017-12-17 at 11 22 37 pm](https://user-images.githubusercontent.com/18646285/34083386-39d622fc-e381-11e7-80c3-0d5b3c531ae9.png)


Is there a possible way to solve this? 
Thank you

**Comment by duytan90** [comment]
About Mental Commands (MC), unless you've loaded a "trained-profile" which contains other mental actions data besides Neutral, or performed training steps (create a training profile, then complete training Neutral, Push, Pull, etc.), you can only get Neutral from MC detection.
Sorry, I can't help you on the crashing issue with only the console log above. Anyways, Let try only call `IEE_EngineDisconnect()` (and also `IEE_EmoStateFree()`, `IEE_EmoEngineEventFree()`) at the very end of the Application instead of before moving back to Main Menu. Or make sure you've cleaned all the Pointers/References properly and didn't access them after clicking on Cancel button.

**Comment by SPineapple** [comment]
I have tried cleaning Pointers/References like you said with IEE_EngineDisconnect(). I am able now exit the window and move to main menu. Thank you.

One last question, I want to make sure if what I'm doing is right.
So after I created a Training profile, then completed the training procedure, I should be able to receive  data other than Neutral. And after I load(`EC_LoadUserProfile`) it, there's no need for training again, right? I also can use `IS_MentalCommandGetCurrentAction` with `IS_MentalCommandGetCurrentActionPower`, and `IS_FacialExpressionIsBlink `?
